packages/web-components/src/components/rux-pop-up-menu/rux-pop-up-menu.tsx

```diff
@@ -1,276 +1,300 @@
-     * to the trigger element where aria-controls === menu id
-     */
-    @Prop({ mutable: true }) anchorEl?: HTMLElement
-    @Watch('triggerEl')
-    @Watch('anchorEl')
-    tieElements() {
-        this._bindElements()
-    }
-
-    /**
-     * Boolean which controls when to show the menu
-     */
-    @Prop({ reflect: true, mutable: true }) open: boolean = false
-    @Watch('open')
-    openMenu() {
-        this._toggleOpenClose()
-    }
+    @Element() el!: HTMLRuxPopUpMenuElement
 
     /**
-     * Emitted when the menu is about to open.
+     * @prop open - determines if the pop up is open or closed
      */
-    @Event({ eventName: 'ruxmenuwillopen' })
-    ruxMenuWillOpen!: EventEmitter<void>
-
+    @Prop({ mutable: true }) open = false
     /**
-     * Emitted when the menu is about to close
+     * @prop placement - the placement of the pop up relative to it's slotted trigger element. A list of
+     * acceptable placements can be found at [floatingui.com/docs](https://floating-ui.com/docs/computePosition#placement)
      */
-    @Event({ eventName: 'ruxmenuwillclose' })
-    ruxMenuWillClose!: EventEmitter<void>
+    @Prop() placement: Placement = 'bottom'
 
-    /**
-     * Emitted when the menu is open.
-     */
-    @Event({ eventName: 'ruxmenudidopen' })
-    ruxMenuDidOpen!: EventEmitter<void>
+    @State() arrowPosition?: string
 
     /**
-     * Emitted when the menu is closed.
+     * @event ruxpopupmenuselected - emits the value of the selected rux-menu-item inside of rux-pop-up-menu
      */
-    @Event({ eventName: 'ruxmenudidclose' })
-    ruxMenuDidClose!: EventEmitter<void>
+    @Event({ eventName: 'ruxpopupmenuselected' })
+    ruxPopUpMenuSelected!: EventEmitter
 
-    componentDidRender() {
+    @Watch('open')
+    handleOpen() {
         if (this.open) {
-            this._setMenuPosition()
-        }
-    }
-
-    connectedCallback() {
-        this._handleClick = this._handleClick.bind(this)
-        this._handleOutsideClick = this._handleOutsideClick.bind(this)
-
-        this._bindElements()
-
-        this._toggleOpenClose()
-    }
-
-    disconnectedCallback() {
-        if (this.triggerEl) {
-            this.triggerEl.removeEventListener('mousedown', this._handleClick)
+            this.content.style.display = 'block'
+            this._startPositioner()
+            window.addEventListener('mousedown', (e: MouseEvent) =>
+                this._handleOutsideClick(e)
+            )
+        } else {
+            this.content.style.display = ''
+            this._stopPositioner()
+            window.removeEventListener('mousedown', (e: MouseEvent) =>
+                this._handleOutsideClick(e)
+            )
         }
     }
 
     /**
-     * Returns 'true' if the menu is open, 'false' if it is not.
+     * @retruns Promise<boolean> - true if the pop-up is open, false if not.
      */
     @Method()
     async isOpen(): Promise<boolean> {
```

Comment:
can we change this to show + hide or show(boolean) ? we need an async way to be able to open and close